# Libraries

In [1]:
suppressWarnings(suppressPackageStartupMessages(library(tidyverse)))
suppressWarnings(suppressPackageStartupMessages(library(haven)))
suppressWarnings(suppressPackageStartupMessages(library(bbplot)))

# Creating new scales in the ILSD dataset

Load the ILSD dataset

In [2]:
ilsd <- read_dta("data/ilsd jun2018 (con scale).dta")

Let's create:
- the logit left right scale
- the "classic" economic scale
- the economic ratio
- the logit economic scale
- the "classic" gal/tan scale
- the gal/tan ratio
- the logit gal/tan scale
- the "classic" economic gal/tan
- the economic gal/tan ratio
- the logit economic gal/tan scale
- a dummy equal to zero when all variables used to create the gal/tan variables are equal to zero

In [3]:
ilsd <- ilsd %>% 
mutate(right = per101 + per104 + per110 + per107 + per401 + per402 + per406 + per410 + per414 + per416 + per601 + per603 + per605 + per609 + per608 + per202,
       left = per102 + per103 + per105 + per106 + per108 + per109 + per403 + per404 + per405 + per408 + per409 + per411 + per412 + per413 + per415 + per417 + per602 + per604 + per606 + per607 + per610 + per611 + per201,
       logit_left_right = log(right + .5) - log(left + .5),
       economic_right = per401 + per402 + per406 + per410 + per414 + per416,
       economic_left = per403 + per404 + per405 + per407 + per408 + per409 + per411 + per412 + per413 + per415 + per417,
       classic_economic = economic_right - economic_left,
       ratio_economic = classic_economic / (economic_right + economic_left),
       logit_economic = log(economic_right + .5) - log(economic_left + .5),
       classic_gal_tan = ((per108 + per608) - (per107 + per607)),
       ratio_gal_tan = classic_gal_tan / (per107 + per108 + per607 + per608),
       logit_gal_tan = log((per108 + per608) + .5) - log((per107 + per607) + .5),
       economic_gal_tan_left = per107 + per405 + per607,
       economic_gal_tan_right = per108 + per406 + per608,
       classic_economic_gal_tan = (economic_gal_tan_right - economic_gal_tan_left),
       ratio_economic_gal_tan = classic_economic_gal_tan / (economic_gal_tan_left + economic_gal_tan_right),
       logit_economic_gal_tan = log(economic_gal_tan_right + .5) - log(economic_gal_tan_left + .5),
       gal_tan_controllo = if_else((per107 == 0 & per108 == 0 & per405 == 0 & per406 == 0 & per607 == 0 & per608 == 0), 1, 0))

The speeches with a null value in the ratio_gal_tan variable have a score equal to zero for all variables used to calculate it (i.e. per107, per108, per607, per608)

In [ ]:
ilsd %>% 
filter(is.na(ratio_gal_tan)) %>% 
select(per107, per108, per607, per608, classic_gal_tan, ratio_gal_tan, logit_gal_tan)

By adding dimension 405 and dimension 406 to the computation of the GAL-TAN scale the number of NAs in the "ratio" version of the scale went down from 173 to 159

In [ ]:
sum(is.na(ilsd$ratio_gal_tan))

In [ ]:
sum(is.na(ilsd$ratio_economic_gal_tan))

## Distributions of the newly created scales

### Logit left-right scale

In [ ]:
ilsd %>% 
ggplot(aes(logit_left_right)) +
geom_histogram(fill = "dark blue") +
bbc_style() +
labs(title = "Frequency distribution for the logit left-right scale") +
theme(plot.title = element_text(size = 20))

In [ ]:
ilsd %>% 
group_by(PARTY) %>% 
summarize(avg_logit_left_right = mean(logit_left_right)) %>% 
arrange(desc(avg_logit_left_right)) %>% 
ggplot(aes(x = fct_reorder(PARTY, avg_logit_left_right), y = avg_logit_left_right, fill = avg_logit_left_right)) +
geom_col() +
coord_flip() +
bbc_style() +
scale_fill_gradient(low = "light blue", high = "dark blue") +
labs(title = "Parties by logit left-right scale") +
theme(plot.title = element_text(size = 20),
      axis.title.x = element_blank(),
      axis.title.y = element_blank(),
      axis.text.x = element_text(size = 9),
      axis.text.y = element_text(size = 7),
      legend.position = "none")

### Logit by economic left-right scale

In [ ]:
ilsd %>% 
ggplot(aes(logit_left_right)) +
geom_histogram(fill = "dark green") +
bbc_style() +
labs(title = "Frequency distribution for the logit \neconomic left-right scale") +
theme(plot.title = element_text(size = 20))

In [ ]:
ilsd %>% 
group_by(PARTY) %>% 
summarize(avg_logit_economic = mean(logit_economic)) %>% 
arrange(desc(avg_logit_economic)) %>% 
ggplot(aes(x = fct_reorder(PARTY, avg_logit_economic), y = avg_logit_economic, fill = avg_logit_economic)) +
geom_col() +
coord_flip() +
bbc_style() +
scale_fill_gradient(low = "light green", high = "dark green") +
labs(title = "Parties by logit economic scale") +
theme(plot.title = element_text(size = 20),
      axis.title.x = element_blank(),
      axis.title.y = element_blank(),
      axis.text.x = element_text(size = 9),
      axis.text.y = element_text(size = 7),
      legend.position = "none") 

### Logit GAL/TAN scale

In [ ]:
ilsd %>% 
ggplot(aes(logit_left_right)) +
geom_histogram(fill = "dark red") +
bbc_style() +
labs(title = "Frequency distribution for the logit GAL-TAN scale") +
theme(plot.title = element_text(size = 20))

In [ ]:
ilsd %>% 
group_by(PARTY) %>% 
summarize(avg_logit_gal_tan = mean(logit_gal_tan)) %>% 
arrange(desc(avg_logit_gal_tan)) %>% 
ggplot(aes(x = fct_reorder(PARTY, avg_logit_gal_tan), y = avg_logit_gal_tan, fill = avg_logit_gal_tan)) +
geom_col() +
coord_flip() +
bbc_style() +
scale_fill_gradient(low = "pink", high = "dark red") +
labs(title = "Parties by logit GAL/TAN scale") +
theme(plot.title = element_text(size = 20),
      axis.title.x = element_blank(),
      axis.title.y = element_blank(),
      axis.text.x = element_text(size = 9),
      axis.text.y = element_text(size = 7),
      legend.position = "none") 

### Logit economic GAL/TAN scale

In [ ]:
ilsd %>% 
ggplot(aes(logit_left_right)) +
geom_histogram(fill = "dark grey") +
bbc_style() +
labs(title = "Frequency distribution for the logit economic GAL-TAN scale") +
theme(plot.title = element_text(size = 18))

In [ ]:
ilsd %>% 
group_by(PARTY) %>% 
summarize(avg_logit_gal_tan = mean(logit_gal_tan)) %>% 
arrange(desc(avg_logit_gal_tan)) %>% 
ggplot(aes(x = fct_reorder(PARTY, avg_logit_gal_tan), y = avg_logit_gal_tan, fill = avg_logit_gal_tan)) +
geom_col() +
coord_flip() +
bbc_style() +
scale_fill_gradient(low = "light gray", high = "black") +
labs(title = "Parties by logit economic GAL/TAN scale") +
theme(plot.title = element_text(size = 20),
      axis.title.x = element_blank(),
      axis.title.y = element_blank(),
      axis.text.x = element_text(size = 9),
      axis.text.y = element_text(size = 7),
      legend.position = "none") 

# Joining the ILSD dataset to the speeches dataset

Load the speeches dataset

In [4]:
load("data/parliamentary_groups2.rds")
texts <- Texts %>% as_tibble()

The ILSD dataset includes the Constituent Assembly which is coded as Legislature n. 1. Let's subtract one from the "Legislature" column

In [5]:
ilsd <- ilsd %>% 
mutate(Legislature_RECODED = Legislature - 1)

The following tibble shows all the parties in the ILSD dataset, their full name (as reported in the codebook), the corresponding party label in the speeches dataset and the legislatures they appear in.

In [ ]:
parties <- tribble(
~party, ~full_name, ~gruppoP, ~gruppoP_recoded, ~legislatures, ~notes,
"AD", "Alleanza Democratica", "PROGR-F", "PROGR-F", "12", "In coalition with 'Alleanza dei Progressisti (PROGR-F)'",
"ALA", "Alleanza Liberalpopolare - Autonomie", NA, NA, "17", "Part of the Mixed Group; in 2016 it became part of 'NCI-SCPI-MAIE'",
"ALP", "Alternativa Libera-Possibile", "SI-SEL-POS-LU", "SI-SEL-POS-LU", "17", "",
"AN", "Alleanza Nazionale", "AN", "AN", "12-15", "",
"API", "Alleanza per l'Italia", "UDCPTP", "UDCPTP", "16", "",
"AUTSVP", "Sudtiroler Volkspartei", NA, NA, "18", "Part of 'Minoranze linguistiche (Gruppo misto)'",
"BNL", "Blocco Nazionale della Libertà", NA, NA , "0", "Our dataset does not include the Constituent Assembly",
"CCD", "Centro Cristiano Democratico", "CCD; CCD-CDU", "UDC", "12-14", "",
"CD", "Centro Democratico", "DES-CD", "DES-CD", "17", "",
"CDU","Cristiani Democratici Uniti", "CCD", "UDC" , "13", "",
"CGM5SLN", "SintesiContrattoGovM5SLN", NA, NA, "18", "",
"CR", "Conservatori e Riformisti", NA, NA, "17", "Part of FI, then it exited it and it was disbanded in 2017",
"DC", "Democrazia Cristiana", "DC; DC-PPI", "DC", "0-11", "",
"DEM", "I Democratici", "DEM-U", "DEM-U", "13", "",
"DL", "Democrazia è Libertà/La Margherita", "MARGH-U", "MARGH-U", "14", "",
"DP", "Democrazia Proletaria", "DP; DP-COM", "DP", "7-10", "",
"DS", "Democratici di Sinistra", "DS-ULIVO; DS-U", "PD-ULIVO", "13-14", "",
"FDI", "Fratelli d'Italia", "FDI-AN; FDI", "FDI", "17-18", "",
"FELD", "Federalisti", "FLD", "FLD", "12", "",
"FI", "Forza Italia", "F-ITA; FI-PDL; FI", "FI-PDL", "12-18", "",
"FLI", "Futuro e Libertà per l’Italia", "FLPTP", "FLPTP", "16", "",
"FUQ", "Fronte dell'Uomo Qualunque", NA, NA, "0", "Our dataset does not include the Constituent Assembly",
"GAL", "Grandi Autonomie e Libertà", NA, NA, "17", "Can't be found in texts; in 2017 it became part of the UDC",
"IDV", "Italia dei Valori", "IDV", "IDV", "15-16", "",
"LEGA", "LEGA", "LNP", "LEGA", "16", "",
"LEU", "Liberi e Uguali" , "LEU", "LEU" , "18", "",
"LN", "Lega Nord", "LEGA; LEGA-N; LEGA-NORD-P; LNA; LNP; LNFP", "LEGA", "10-18", "",
"LV", "Liga Veneta", NA, NA, "9", "Today it's part of Lega, but it is absent in texts for the 9th legislature",
"M5S", "Movimento 5 stelle", "M5S", "M5S", "17-18", "",
"MPA", "Movimento per le Autonomie", NA, NA, "16", "Part of the right-wing government coalition",
"MSI", "Movimento Sociale Italiano", "MSI-DN; MSI", "MSI", "1-11", "",
"NCD", "Nuovo Centro Destra", "AP-CPE-NCD-NCI", "AP-CPE-NCD-NCI", "17", "",
"NPSI", "Nuovo Partito Socialista Italiano", NA, NA, "14", "texts includes DCA-NPSI. However, there's no match for the 14th legislature",
"NS", "Noi Sud", "NOI SUD-LIBERTA\' ED AUTONOMIA/POPOLARI D\'ITALIA DOMANI-PID/MOVIMENTO DI RESPONSABILITA\' NAZIONALE-MRN/AZIONE POPOLARE/ALLEANZA DI CENTRO-ADC/INTESA POPOLARE", "NOI SUD-LIBERTA\' ED AUTONOMIA/POPOLARI D\'ITALIA DOMANI-PID/MOVIMENTO DI RESPONSABILITA\' NAZIONALE-MRN/AZIONE POPOLARE/ALLEANZA DI CENTRO-ADC/INTESA POPOLARE", "16", "",
"PAT", "Patto Segni", NA, NA, "12", "Absent in texts",
"PCI", "Partito Comunista Italiano", "PCI", "PCI", "0-10", "",
"PD", "Partito Democratico", "PD-U; PD", "PD-ULIVO", "15-18", "",
"PDCI", "Partito dei Comunisti Italiani", "COM/IT/", "COM/IT/", "13-15", "",
"PDIUM", "Partito Democratico Italiano di Unità Monarchica", "PDIUM", "PDIUM", "3-5", "",
"PDL", "Popolo Delle Libertà", "PDL; FI-PDL", "FI-PDL", "16-17", "",
"PDS", "Partito Democratico della Sinistra", "PDS", "PDS", "10-13", "",
"PI", "Per l'Italia", "DES-CD", "DES-CD", "17", "",
"PLI", "Partito Liberale Italiano", "PLI", "PLI", "0-11", "",
"PMP", "Partito Monarchico Popolare", "PMP", "PMP", "2-3", "",
"PNM", "Partito Nazionale Monarchico", "PNM", "PNM", "1-3", "",
"PPI", "Partito Popolare Italiano", "PPI", "PPI", "12-13", "",
"PR", "Partito Radicale", "RADICALE", "RADICALE", "7-11", "",
"PRI", "Partito Repubblicano Italiano", "PRI", "PRI", "0-11", "",
"PSDI", "Partito Socialista Democratico Italiano", "PSI-PSDI; PSI", "PSDI", "0-11", "",
"PSI", "Partito Socialista Italiano", "PSI-PSDI; PSI", "PSI", "0-12", "",
"PSIUP", "Partito Socialista di Unità Proletaria", "PSIUP", "PSIUP", "4-5", "",
"PSU", "Partito Socialista Unitario", NA, NA, "5", "The codebook refers to this party as 'Partito Socialista Unitario, but in fact it's probably 'Partito Socialista Unificato'. Anyway, it's not in texts. However, it could refer to both PSI and PSDI",
"PT", "Popolo e Territorio", "NOI SUD-LIBERTA' ED AUTONOMIA/POPOLARI D'ITALIA DOMANI-PID/MOVIMENTO DI RESPONSABILITA' NAZIONALE-MRN/AZIONE POPOLARE/ALLEANZA DI CENTRO-ADC/INTESA POPOLARE", "NOI SUD-LIBERTA' ED AUTONOMIA/POPOLARI D'ITALIA DOMANI-PID/MOVIMENTO DI RESPONSABILITA' NAZIONALE-MRN/AZIONE POPOLARE/ALLEANZA DI CENTRO-ADC/INTESA POPOLARE", "16", "",
"RC", "Rifondazione Comunista", "RC-PROGR; RC-SE; RC", "RC", "11-15", "",
"RETE", "La Rete", "RETE", "RETE", "11-12", "",
"RI", "Rinnovamento Italiano",  "RINN/IT", "RINN/IT", "13", "", 
"RNP", "Rosa Nel Pugno", "SOCRAD-RNP", "SOCRAD-RNP", "15", "",
"SC", "Scelta Civica", "NCI-SCPI-MAIE", "NCI-SCPI-MAIE", "17", "", 
"SD", "Sinistra Democratica", "SDPSE", "SDPSE", "15", "Mostly MPs from DS who did not join PD. It unified several left-wing parties",
"SDI", "Socialisti Democratici Italiani", NA, NA, "13-14", "It was the direct continuation of the Italian Socialists",
"SEL", "Sinistra Ecologia e Libertà", "SI-SEL-POS-LU", "SI-SEL-POS-LU", "17", "",
"UDC", "Unione dei Democratici Cristiani", "UNIONE DEI DEMOCRATICI CRISTIANI E DEI DEMOCRATICI DI CENTRO; CCD; CCD-CDU; UDCPTP", "UDC", "14-17", "",
"UDEUR", "Unione Democratici per l'Europa", "UDEUR", "UDEUR", "13-15", "", 
"UDR", "Unione Democratica per la Repubblica", "UDR", "UDR", "13", "",
"ULIVO", "Ulivo", "PD-U", "PD-ULIVO", "15", "",
"VER", "Verdi", "VERDE; VERDI", "VERDI", "10-15", "")

writexl::write_xlsx(x = parties, "data/ilsd_parties.xlsx")

Recode the 'gruppoP' variable in the speeches dataset by grouping together labels belonging to the same party. This will simplify the merging of the two datasets

In [6]:
texts <- texts %>% 
mutate(gruppoP_recoded = case_when(
gruppoP %in% c('CCD', 'CCD-CDU', 'UNIONE DEI DEMOCRATICI CRISTIANI E DEI DEMOCRATICI DI CENTRO', 'UDCPTP')  ~ 'UDC',
gruppoP == 'COMUNISTA' ~ 'RC',
gruppoP == 'DC-PPI' ~ 'DC',
gruppoP %in% c('DP-COM', 'PDUP-DP') ~ 'DP',
gruppoP == "FED/EUR/" ~ 'RADICALE',
gruppoP == 'FDI-AN' ~ 'FDI',
gruppoP %in% c('F-ITA', 'FI', 'PDL') ~ 'FI-PDL',
gruppoP == 'GC-PDS' ~ "PDS",
gruppoP %in% c('LEGA-N', 'LEGA-NORD-P', 'LNA', 'LNFP', 'LNP') ~ 'LEGA',
gruppoP == 'MSI-DN' ~ "MSI",
gruppoP %in% c('DS-U', 'PD-U', 'PD-ULIVO', 'PD', 'DS-ULIVO') ~ 'PD-ULIVO',
gruppoP == 'PDI' ~ 'PDIUM',
gruppoP == 'POP-UDEUR' ~ 'UDEUR',
gruppoP == 'PSI-PSDI' ~ "PSI",
gruppoP %in% c('RC-PROGR', 'RC-SE') ~ 'RC',
gruppoP == 'VERDE' ~ 'VERDI',
TRUE ~ gruppoP))

The following are the parties in the ILSD dataset that I couldn't identify in the speeches dataset

In [ ]:
parties %>% 
filter(is.na(gruppoP))

Recode the 'PARTY' variable from the ILSD dataset to make the party labels identical to the ones in the speeches dataset

In [7]:
ilsd <- ilsd %>% 
mutate(PARTY_RECODED = case_when(
PARTY == "AD" ~ "PROGR-F",
PARTY == "ALP" ~ 'SI-SEL-POS-LU',
PARTY == "API" ~ "UDCPTP",
PARTY == "CCD" ~ 'UDC',
PARTY == "CD" ~ "DES-CD",
PARTY == "CDU" ~ "UDC",
PARTY == "DEM" ~ "DEM-U",
PARTY == "DL" ~ "MARGH-U",
PARTY == "DS" ~ "PD-ULIVO",
PARTY == "FELD" ~ "FLD",
PARTY == "FI" ~ "FI-PDL",
PARTY == "FLI" ~ "FLPTP",
PARTY == "LN" ~ "LEGA",
PARTY == "NCD" ~ "AP-CPE-NCD-NCI",
PARTY == "NS" ~ "NOI SUD-LIBERTA' ED AUTONOMIA/POPOLARI D'ITALIA DOMANI-PID/MOVIMENTO DI RESPONSABILITA' NAZIONALE-MRN/AZIONE POPOLARE/ALLEANZA DI CENTRO-ADC/INTESA POPOLARE",
PARTY == "PD" ~ "PD-ULIVO",
PARTY == "PDCI" ~ "COM/IT/",
PARTY == "PDL" ~ "FI-PDL",
PARTY == "PI" ~ "DES-CD",
PARTY == "PR" ~ "RADICALE",
PARTY == "PT" ~ "NOI SUD-LIBERTA' ED AUTONOMIA/POPOLARI D'ITALIA DOMANI-PID/MOVIMENTO DI RESPONSABILITA' NAZIONALE-MRN/AZIONE POPOLARE/ALLEANZA DI CENTRO-ADC/INTESA POPOLARE",
PARTY == "RI" ~ "RINN/IT",
PARTY == "RNP" ~ "SOCRAD-RNP",
PARTY == "SC" ~ "NCI-SCPI-MAIE",
PARTY == "SD" ~ "SDPSE",
PARTY == "SEL" ~ "SI-SEL-POS-LU",
PARTY == "ULIVO" ~ "PD-ULIVO",
PARTY == "VER" ~ "VERDI",
TRUE ~ PARTY
)) 

Extract the year from the 'Edate' variable with a regex

In [8]:
ilsd <- ilsd %>% 
mutate(Year = as.integer(str_extract(string = Edate, pattern = "\\d{4}")))

Parties often have more than one row for each party-legislature-year combination. Let's compute the mean for each of the variables we want to merge in the speeches dataset.

In [9]:
ilsd <- ilsd %>% 
group_by(PARTY_RECODED, Legislature_RECODED, Year) %>% 
mutate(
    across(.cols = c("left_right", "ratio_leftright", "logit_left_right", "classic_economic", "ratio_economic", 
                     "logit_economic", "classic_gal_tan", "ratio_gal_tan", "logit_gal_tan", "classic_economic_gal_tan", 
                     "ratio_economic_gal_tan", "logit_economic_gal_tan", "gal_tan_controllo"),
           .fns = ~ mean(., na.rm = TRUE)))

In [10]:
nrow(ilsd)

[1] 761

Let's filter the ILSD dataset to keep only unique combinations of party, legislature and year. If we don't do this, the left join with the texts dataset will output more rows than we have in the left dataset.

In [11]:
ilsd <- distinct(ilsd, PARTY_RECODED, Legislature_RECODED, Year, .keep_all = TRUE)

Now we only have 612 rows

In [12]:
nrow(ilsd)

[1] 612

Left join the ILSD dataset to our speeches dataset

In [13]:
joined_texts <- texts %>% 
left_join(ilsd, by = c("gruppoP_recoded" = "PARTY_RECODED", "legislature" = "Legislature_RECODED", "year" = "Year"))

The joined dataset has the same number of rows as the previous dataset

In [14]:
nrow(texts) == nrow(joined_texts)

[1] TRUE

The ILSD dataset contains fewer years compared to 'texts'. This generated a large number of null values in the joined dataset. 

In [15]:
map_dfr(joined_texts, ~ sum(is.na(.)) / length(.)) %>% 
select(left_right, ratio_leftright, logit_left_right, classic_economic, ratio_economic, logit_economic, classic_gal_tan,
      ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, ratio_economic_gal_tan, logit_economic_gal_tan, gal_tan_controllo)

left_right ratio_leftright logit_left_right classic_economic ratio_economic
1 0.4179075  0.4179075       0.4179075        0.4179075        0.4236818     
  logit_economic classic_gal_tan ratio_gal_tan logit_gal_tan
1 0.4179075      0.4179075       0.523161      0.4179075    
  classic_economic_gal_tan ratio_economic_gal_tan logit_economic_gal_tan
1 0.4179075                0.5073761              0.4179075             
  gal_tan_controllo
1 0.4179075

Let's fill those values by using the previous year as reference.

In [16]:
joined_texts <- joined_texts %>% 
arrange(year) %>% 
group_by(gruppoP_recoded, legislature) %>% 
fill_(fill_cols = c("left_right", "ratio_leftright", "logit_left_right", "classic_economic", "ratio_economic", 
                    "logit_economic", "classic_gal_tan", "ratio_gal_tan", "logit_gal_tan", "classic_economic_gal_tan", 
                    "ratio_economic_gal_tan", "logit_economic_gal_tan", "gal_tan_controllo"),
       .direction = "down")

The % of null values went down from 41.8% to 10.2%

In [17]:
map_dfr(joined_texts, ~ sum(is.na(.)) / length(.)) %>% 
select(left_right, ratio_leftright, logit_left_right, classic_economic, ratio_economic, logit_economic, classic_gal_tan,
      ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, ratio_economic_gal_tan, logit_economic_gal_tan, gal_tan_controllo)

left_right ratio_leftright logit_left_right classic_economic ratio_economic
1 0.1017055  0.1017055       0.1017055        0.1017055        0.1078197     
  logit_economic classic_gal_tan ratio_gal_tan logit_gal_tan
1 0.1017055      0.1017055       0.2213797     0.1017055    
  classic_economic_gal_tan ratio_economic_gal_tan logit_economic_gal_tan
1 0.1017055                0.1988006              0.1017055             
  gal_tan_controllo
1 0.1017055

Saving the joined and transformed dataset as an .rds file

In [ ]:
saveRDS(joined_texts, file = "data/joined_texts.rds")

# Filling the NULL values

In [18]:
joined_texts %>% 
filter(is.na(left_right)) %>% 
distinct(gruppoP_recoded)

legislature
1   1         
2   1         
3   1         
4   2         
5   2         
6   3         
7   3         
8   4         
9   4         
10  5         
11  5         
12  5         
13  6         
14  7         
15  7         
16  8         
17  8         
18  9         
19  9         
20 10         
21 10         
22 10         
23 11         
24 11         
25 12         
26 12         
27 12         
28 12         
29 13         
30 13         
31 13         
32 13         
33 13         
34 14         
35 15         
36 15         
37 15         
38 16         
39 16         
40 16         
41 17         
42 17         
43 17         
44 18         
45 18         
   gruppoP_recoded                                                                                                                                             
1  MISTO                                                                                                                                                       
2  US                                                                                                                                                          
3  PSU                                                                                                                                                         
4  MISTO                                                                                                                                                       
5  PMP                                                                                                                                                         
6  PDIUM                                                                                                                                                       
7  MISTO                                                                                                                                                       
8  PSIUP                                                                                                                                                       
9  MISTO                                                                                                                                                       
10 MISTO                                                                                                                                                       
11 PSI                                                                                                                                                         
12 PSDI                                                                                                                                                        
13 MISTO                                                                                                                                                       
14 MISTO                                                                                                                                                       
15 CD-DN                                                                                                                                                       
16 MISTO                                                                                                                                                       
17 PDUP                                                                                                                                                        
18 MISTO                                                                                                                                                       
19 SI                                                                                                                                                          
20 SIN/IND/                                                                                                      

In [88]:
tribble(~legislatura, ~gruppoP_recoded, ~nome_completo, ~nota,
        1, "MISTO", "", "",
        1, "US", "Unità Socialista", "Nacque da una scissione del Partito Socialista Italiano",
        1, "PSU", "Partito Socialista Unitario", "Nato da fusione tra Movimento Socialista Autonomista, l'Unione dei Socialisti e la corrente di sinistra uscita dal Partito Socialista dei Lavoratori Italiani",
        2, "MISTO", "", "",
        2, "PMP", "Partito Monarchico Popolare", "Valori nulli per 1953 e 1954",
        3, "PDIUM", "Partito Democratico Italiano di Unità Monarchica", "Valori nulli per 1958 e 1959",
        3, "MISTO", "", "",
        4, "PSIUP", "Partito Socialista Italiano di Unità Proletaria", "Valori nulli per 1963",
        4, "MISTO", "", "",
        5, "MISTO", "", "",
        5, "PSI", "Partito Socialista Italiano", "Valori nulli per il 1968",
        5, "PSDI", "Partito Socialista Democratico Italiano", "Valori nulli per il 1968",
        6, "MISTO", "", "",
        7, "MISTO", "", "",
        7, "CD-DN", "Democrazia Nazionale - Costituente di Destra", "nato nel febbraio 1977 dall'omonima corrente moderata del MSI-DN, e durato sino allo scioglimento nel 1979",
        8, "MISTO", "", "", 
        8, "PDUP", "Partito di Unità Proletaria per il Comunismo", "Coalizzato con DP",
        9, "MISTO", "", "",
        9, "SI", "Sinistra Indipendente", "candidati contigui al PCI a partire dal 1967-68 fino al suo scioglimento, ma per provenienza e idee erano esterni alla struttura del partito",
        10, "SIN/IND/", "Sinistra Indipendente", "candidati contigui al PCI a partire dal 1967-68 fino al suo scioglimento, ma per provenienza e idee erano esterni alla struttura del partito",
        10, "MISTO", "", "",
        10, "PDS", "Partito Democratico della Sinistra", "Valori nulli per 1987, 1988, 1989, 1990",
        11, "MISTO", "", "",
        11, "UDC", "Centro Cristiano Democratico", "Fondato da esponenti moderati della DC",
        12, "DEMO", "Patto dei Democratici", "lista elettorale costituitasi in vista delle elezioni regionali del 1995 da Alleanza Democratica, Patto Segni e Socialisti Italiani",
        12, "MISTO", "", "",
        12, "FLD", "Federalisti e Liberaldemocratici", "In texts è presente solo nel 1994, in ILSD solo nel 1995",
        12, "LIFED", "Lega Italiana Federalista", "Il partito fu fondato il 13 febbraio 1995 da alcuni esponenti della Lega Nord",
        13, "PD-ULIVO", "PDS - L'Ulivo / Democratici di Sinistra - L'Ulivo", "Valori nulli per 1996 e 1997. Usare valori PDS del 1996?",
        13, "UDEUR", "Popolari UDEUR", "Nato nel 1999; valori nulli per 1996, 1997 e 1998",
        13, "MISTO", "", "",
        13, "UDR", "Unione Democratica per la Repubblica", "Fondata nel 1998 da elementi proveniente da CDU, CDR e PSDI; valori nulli per 1996 e 1997",
        13, "DEM-U", "I Democratici - L'Ulivo", "Valori nulli per 1996, 1997 e 1998",
        14, "MISTO", "", "",
        15, "SDPSE", "Sinistra Democratica Per il Socialismo Europeo", "Valori nulli per 2006 e 2007",
        15, "MISTO", "", "",
        15, "DCA-NPSI", "Democrazia Cristiana per le Autonomie-Nuovo Partito Socialista Italiano", "Assente in ILSD per ragioni non identificate",
        16, "FLPTP", "Futuro e Libertà Per il Terzo Polo", "In ILSD c'è Unione di Centro Per il Terzo Polo",
        16, "NOI SUD-LIBERTA' ED AUTONOMIA/POPOLARI D'ITALIA DOMANI-PID/MOVIMENTO DI RESPONSABILITA' NAZIONALE-MRN/AZIONE POPOLARE/ALLEANZA DI CENTRO-ADC/INTESA POPOLARE", "", "Valori nulli per 2008 e 2009",
        16, "MISTO", "", "",
        17, "MDP-LU", "ARTICOLO 1-MOVIMENTO DEMOCRATICO E PROGRESSISTA", "Assente in ILSD; ci sono però Alternativa Libera-Possibile e SEL",
        17, "MISTO", "", "",
        17, "CI", "Civici e Innovatori", "fino al 2016 noto come Scelta Civica per l'Italia",
        18, "MISTO", "", "",      
        18, "IV", "", "Assente in ILSD"
        
       ) %>% 
filter(gruppoP_recoded != "MISTO")

legislatura
1   1         
2   1         
3   2         
4   3         
5   4         
6   5         
7   5         
8   7         
9   8         
10  9         
11 10         
12 10         
13 11         
14 12         
15 12         
16 12         
17 13         
18 13         
19 13         
20 13         
21 15         
22 15         
23 16         
24 16         
25 17         
26 17         
27 18         
   gruppoP_recoded                                                                                                                                             
1  US                                                                                                                                                          
2  PSU                                                                                                                                                         
3  PMP                                                                                                                                                         
4  PDIUM                                                                                                                                                       
5  PSIUP                                                                                                                                                       
6  PSI                                                                                                                                                         
7  PSDI                                                                                                                                                        
8  CD-DN                                                                                                                                                       
9  PDUP                                                                                                                                                        
10 SI                                                                                                                                                          
11 SIN/IND/                                                                                                                                                    
12 PDS                                                                                                                                                         
13 UDC                                                                                                                                                         
14 DEMO                                                                                                                                                        
15 FLD                                                                                                                                                         
16 LIFED                                                                                                                                                       
17 PD-ULIVO                                                                                                                                                    
18 UDEUR                                                                                                                                                       
19 UDR                                                                                                                                                         
20 DEM-U                                                                                                                                                       
21 SDPSE                                                                                                                                                       
22 DCA-NPSI                                                    

## Unità Socialista (US) and Partito Socialista Unitario (PSU)

"Unità Socialista" was born out of an agreement between PSDI (then "Partito Socialista dei Lavoratori Italiani") and "Unione dei Socialisti". The latter splintered from PSI. 

"Partito Socialista Unitario" was founded in 1949 as a result of the merging of three parties: Movimento Socialista Autonomista, Unione dei Socialisti and PSDI (then "Partito Socialista dei Lavoratori Italiani"). 

Movimento Socialista Autonomista is not available in ILSD. I am thus filling the null values for both US and PSU with the average between PSDI and PSI. 

In [20]:
ilsd %>% 
filter(Legislature_RECODED == 1 & (PARTY_RECODED == "PSI" | PARTY_RECODED == "PSDI")) %>% 
group_by(Year) %>% 
summarize(left_right = mean(left_right),
          ratio_leftright = mean(logit_left_right),
          logit_left_right = mean(logit_left_right),
         classic_economic = mean(classic_economic),
         ratio_economic = mean(ratio_economic),
         logit_economic = mean(logit_economic),
         classic_gal_tan = mean(classic_gal_tan),
         ratio_gal_tan = mean(ratio_gal_tan),
         logit_gal_tan = mean(logit_gal_tan),
         classic_economic_gal_tan = mean(classic_economic_gal_tan),
         ratio_economic_gal_tan = mean(ratio_economic_gal_tan),
         logit_economic_gal_tan = mean(logit_economic_gal_tan))

Year left_right ratio_leftright logit_left_right classic_economic
1 1948 -31.98393  -2.899208       -2.899208        -14.01744       
2 1949 -31.98393  -2.899208       -2.899208        -14.01744       
3 1950 -24.45500  -2.585112       -2.585112        -13.48500       
4 1951 -25.60750  -2.243195       -2.243195        -15.92500       
  ratio_economic logit_economic classic_gal_tan ratio_gal_tan logit_gal_tan
1 -0.9098242     -2.577971       0.48207          0            0.3934124   
2 -0.9098242     -2.577971       0.48207          0            0.3934124   
3 -0.8469818     -2.413050      -0.07500        NaN           -0.1311821   
4 -0.9035449     -2.740863      -0.30000         -1           -0.4237901   
  classic_economic_gal_tan ratio_economic_gal_tan logit_economic_gal_tan
1  0.48207                   0                     0.3934124            
2  0.48207                   0                     0.3934124            
3 -0.07500                 NaN                    -0.1311821            
4 -0.30000                  -1                    -0.4237901

In [21]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ -31.98393,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -24.45500,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -25.60750,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ -2.899208,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -2.585112,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -2.243195,
TRUE ~ ratio_leftright),
      logit_left_right = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ -2.899208,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -2.585112,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -2.243195,
TRUE ~ logit_left_right),
      classic_economic = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ -14.01744,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -13.48500,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -15.92500,
TRUE ~ classic_economic),
      ratio_economic = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ -0.9098242,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -0.8469818,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -0.9035449,
TRUE ~ ratio_economic),
      logit_economic = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ -2.577971,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -2.413050,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -2.740863,
TRUE ~ logit_economic),
      classic_gal_tan = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ 0.48207,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -0.07500,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -0.30000,
TRUE ~ classic_gal_tan),
      ratio_gal_tan = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ 0,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -1,
TRUE ~ ratio_gal_tan),
      logit_gal_tan = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ 0.3934124,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -0.1311821,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -0.4237901,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ 0.48207,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -0.07500,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -0.30000,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ 0,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -1,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1948 ~ 0.3934124,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1950 ~ -0.1311821,
gruppoP_recoded %in% c('US', 'PSU') & legislature == 1 & year == 1951 ~ -0.4237901,
TRUE ~ logit_economic_gal_tan)) 

## Partito Monarchico Popolare (PMP)

It was founded in 1954 by a split from Partito Nazionale Monarchico. I am thus filling the null values for PMP with values for PNM

In [23]:
ilsd %>% 
filter(Legislature_RECODED == 2 & PARTY_RECODED == "PNM" & Year %in% c(1953, 1954)) %>% 
select(left_right, ratio_leftright, logit_left_right, classic_economic, ratio_economic, logit_economic, 
      classic_gal_tan, ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, ratio_economic_gal_tan, logit_economic_gal_tan)

Adding missing grouping variables: `PARTY_RECODED`, `Legislature_RECODED`, `Year`



PARTY_RECODED Legislature_RECODED Year left_right ratio_leftright
1 PNM           2                   1953  6.178276  0.1603617      
2 PNM           2                   1954 12.906653  0.3385695      
  logit_left_right classic_economic ratio_economic logit_economic
1 0.3151961        -0.09371656      -0.02436869    -0.04303556   
2 0.7013311        -5.97460102      -0.57692307    -1.65142326   
  classic_gal_tan ratio_gal_tan logit_gal_tan classic_economic_gal_tan
1 -0.9362835      -0.6          -0.8278213    -0.9362835              
2 -0.4273504      -1.0          -0.4983639     0.1424501              
  ratio_economic_gal_tan logit_economic_gal_tan
1 -0.6000000             -0.82782131           
2  0.1428571              0.09543616

In [24]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ 6.178276,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ 12.906653,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ 0.1603617,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ 0.3385695,
TRUE ~ ratio_leftright),
       logit_left_right = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ 0.3151961,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ 0.7013311,
TRUE ~ logit_left_right),
       classic_economic = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.09371656,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ -5.97460102,
TRUE ~ classic_economic),
       ratio_economic = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.02436869,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ -0.57692307,
TRUE ~ ratio_economic),
       logit_economic = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.04303556, 
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ -1.65142326,
TRUE ~ logit_economic),
       classic_gal_tan = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.9362835,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ -0.4273504,
TRUE ~ classic_gal_tan),
       ratio_gal_tan = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.6,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ -1.0,
TRUE ~ ratio_gal_tan),
       logit_gal_tan = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.8278213,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ -0.4983639,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.9362835,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ 0.1424501,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.6000000,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ 0.1428571,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded == "PMP" & legislature == 2 & year == 1953 ~ -0.82782131,
gruppoP_recoded == "PMP" & legislature == 2 & year == 1954 ~ 0.09543616,
TRUE ~ logit_economic_gal_tan))

## Partito Democratico Italiano di Unità Monarchica (PDIUM)

It was founded in 1959 as a result of the merging between Partito Nazionale Monarchico (PNM) and Partito Monarchico Popolare (PMP). I'll thus take the average of those two parties and use those values to fill the NAs.

In [29]:
ilsd %>% 
filter(Legislature_RECODED == 3 & (PARTY_RECODED == "PNM" | PARTY_RECODED == "PMP")) %>% 
group_by(Year) %>% 
summarize(left_right = mean(left_right),
          ratio_leftright = mean(logit_left_right),
          logit_left_right = mean(logit_left_right),
         classic_economic = mean(classic_economic),
         ratio_economic = mean(ratio_economic),
         logit_economic = mean(logit_economic),
         classic_gal_tan = mean(classic_gal_tan),
         ratio_gal_tan = mean(ratio_gal_tan),
         logit_gal_tan = mean(logit_gal_tan),
         classic_economic_gal_tan = mean(classic_economic_gal_tan),
         ratio_economic_gal_tan = mean(ratio_economic_gal_tan),
         logit_economic_gal_tan = mean(logit_economic_gal_tan))

Year left_right ratio_leftright logit_left_right classic_economic
1 1958 16.075050  0.8142263       0.8142263        18.0120900      
2 1959  2.450408  0.1127485       0.1127485        -0.7208499      
  ratio_economic logit_economic classic_gal_tan ratio_gal_tan logit_gal_tan
1 0.55125044     1.35203540     -3.488119       -1            -1.9676145   
2 0.02636363     0.04431227     -0.735303       -1            -0.8847821   
  classic_economic_gal_tan ratio_economic_gal_tan logit_economic_gal_tan
1 -3.488119                -1                     -1.9676145            
2 -0.735303                -1                     -0.8847821

In [30]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ 16.075050,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ 2.450408,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ 0.8142263,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ 0.1127485,
TRUE ~ ratio_leftright),
       logit_left_right = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ 0.8142263,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ 0.1127485,
TRUE ~ logit_left_right),
       classic_economic = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ 18.0120900,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ -0.7208499,
TRUE ~ classic_economic),
       ratio_economic = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ 0.55125044,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ 0.02636363,
TRUE ~ ratio_economic),
       logit_economic = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ 1.35203540, 
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ 0.04431227,
TRUE ~ logit_economic),
       classic_gal_tan = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ -3.488119,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ -0.735303,
TRUE ~ classic_gal_tan),
       ratio_gal_tan = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ -1,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ -1,
TRUE ~ ratio_gal_tan),
       logit_gal_tan = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ -1.9676145,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ -0.8847821,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ -3.488119,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ -0.735303,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ -1,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ -1,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1958 ~ -1.9676145,
gruppoP_recoded == "PDIUM" & legislature == 3 & year == 1959 ~ -0.8847821,
TRUE ~ logit_economic_gal_tan))

## Partito Socialista Italiano di Unità Proletaria

It was founded in 1964 by a split from Partito Socialista Italiano. I'll use the values from PSI to fill the NAs.

In [32]:
ilsd %>% 
filter(Legislature_RECODED == 4 & Year == 1963 & PARTY_RECODED == "PSI") %>% 
select(left_right, ratio_leftright, logit_left_right, classic_economic, ratio_economic, logit_economic, 
      classic_gal_tan, ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, ratio_economic_gal_tan, logit_economic_gal_tan)

Adding missing grouping variables: `PARTY_RECODED`, `Legislature_RECODED`, `Year`



PARTY_RECODED Legislature_RECODED Year left_right ratio_leftright
1 PSI           4                   1963 -35.595    -0.7446428     
  logit_left_right classic_economic ratio_economic logit_economic
1 -1.87759         -24.17           -0.9420962     -3.059421     
  classic_gal_tan ratio_gal_tan logit_gal_tan classic_economic_gal_tan
1 -0.28           -0.5090909    -0.2732718    -0.28                   
  ratio_economic_gal_tan logit_economic_gal_tan
1 -0.5090909             -0.2732718

In [33]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -35.595,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -0.7446428,
TRUE ~ ratio_leftright),
       logit_left_right = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -1.87759,
TRUE ~ logit_left_right),
       classic_economic = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -24.17,
TRUE ~ classic_economic),
       ratio_economic = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -0.9420962,
TRUE ~ ratio_economic),
       logit_economic = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -3.059421, 
TRUE ~ logit_economic),
       classic_gal_tan = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -0.28,
TRUE ~ classic_gal_tan),
       ratio_gal_tan = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -0.5090909,
TRUE ~ ratio_gal_tan),
       logit_gal_tan = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -0.2732718,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -0.28,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -0.5090909,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded == "PSIUP" & legislature == 4 & year == 1963 ~ -0.2732718,
TRUE ~ logit_economic_gal_tan))

## Partito Socialista Italiano - Partito Socialista Democratico Italiano

In 1968 PSI and PSDI were both part of Partito Socialista Unificato

In [43]:
ilsd %>% 
filter(Legislature_RECODED == 5 & PARTY_RECODED == 'PSU' & Year == 1968) %>% 
select(left_right, ratio_leftright, logit_left_right, classic_economic, ratio_economic, logit_economic, 
      classic_gal_tan, ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, ratio_economic_gal_tan, logit_economic_gal_tan)

Adding missing grouping variables: `PARTY_RECODED`, `Legislature_RECODED`, `Year`



PARTY_RECODED Legislature_RECODED Year left_right ratio_leftright
1 PSU           5                   1968 -21.075    -0.430095      
  logit_left_right classic_economic ratio_economic logit_economic
1 -0.9760826       -19.955          -0.9149279     -2.710567     
  classic_gal_tan ratio_gal_tan logit_gal_tan classic_economic_gal_tan
1 -4.25           -1            -2.095158     -4.25                   
  ratio_economic_gal_tan logit_economic_gal_tan
1 -1                     -2.095158

In [44]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -21.075,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -0.430095,
TRUE ~ ratio_leftright),
       logit_left_right = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -0.9760826,
TRUE ~ logit_left_right),
       classic_economic = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -19.955,
TRUE ~ classic_economic),
       ratio_economic = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -0.9149279,
TRUE ~ ratio_economic),
       logit_economic = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -2.710567, 
TRUE ~ logit_economic),
       classic_gal_tan = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -4.25,
TRUE ~ classic_gal_tan),
       ratio_gal_tan = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -1,
TRUE ~ ratio_gal_tan),
       logit_gal_tan = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -2.095158,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -4.25,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -1,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded %in% c('PSI', 'PSDI') & legislature == 5 & year == 1968 ~ -2.095158,
TRUE ~ logit_economic_gal_tan))

## Democrazia Nazionale - Costituente di Destra (CD-DN)

In [54]:
ilsd %>% 
filter(Legislature_RECODED == 7 & PARTY_RECODED == 'MSI') %>% 
select(Legislature_RECODED, PARTY_RECODED, Year, left_right, ratio_leftright, logit_left_right, classic_economic, 
       ratio_economic, logit_economic, classic_gal_tan, ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, 
       ratio_economic_gal_tan, logit_economic_gal_tan)


Legislature_RECODED PARTY_RECODED Year left_right ratio_leftright
1 7                   MSI           1976 10.73986   0.2586207      
2 7                   MSI           1978 55.87906   0.9896504      
3 7                   MSI           1979 23.28817   0.8315191      
  logit_left_right classic_economic ratio_economic logit_economic
1 0.5162467        0.3579952        0.4285714      0.3951808     
2 4.2702242        1.1687500        1.0000000      1.2052221     
3 2.2131744        4.0537131        0.5693994      1.0966668     
  classic_gal_tan ratio_gal_tan logit_gal_tan classic_economic_gal_tan
1 -0.8353222       -1           -0.9823198    -0.8353222              
2  0.0000000      NaN            0.0000000     0.0000000              
3 -0.3324468       -1           -0.5097612    -0.3324468              
  ratio_economic_gal_tan logit_economic_gal_tan
1  -1                    -0.9823198            
2 NaN                     0.0000000            
3  -1                    -0.5097612

In [55]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ 10.73986,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 55.87906,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ 23.28817,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ 0.2586207,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 0.9896504,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ 0.8315191,
TRUE ~ ratio_leftright),
      logit_left_right = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ 0.5162467,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 4.2702242,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ 2.2131744,
TRUE ~ logit_left_right),
      classic_economic = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ 0.3579952,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 1.1687500,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ 4.0537131,
TRUE ~ classic_economic),
      ratio_economic = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ 0.4285714,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 1.0000000,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ 0.5693994,
TRUE ~ ratio_economic),
      logit_economic = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ 0.3951808,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 1.2052221,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ 1.0966668,
TRUE ~ logit_economic),
      classic_gal_tan = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ -0.8353222,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 0.0000000,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ -0.3324468,
TRUE ~ classic_gal_tan),
      ratio_gal_tan = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ -1,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ -1,
TRUE ~ ratio_gal_tan),
      logit_gal_tan = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ -0.9823198,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 0.0000000,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ -0.5097612,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ -0.8353222,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 0.0000000,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ -0.3324468,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ -1,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ -1,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1976 ~ -0.9823198,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1978 ~ 0.0000000,
gruppoP_recoded == "CD-DN" & legislature == 7 & year == 1979 ~ -0.5097612,
TRUE ~ logit_economic_gal_tan)) 

## Partito di Unità Proletaria per il Comunismo (PdUP)

It was founded in 1974 with the merge between "Partito di Unità Proletaria" and "Il Manifesto" group. Neither of those two parties are in ILSD in the 8th legislature, so I'll fill the null values with the values pertaining to Democrazia Proletaria (DP), which PdUP was in coalition with.

In [72]:
ilsd %>% 
filter(Legislature_RECODED == 8 & PARTY_RECODED == "DP") %>% 
select(Legislature_RECODED, PARTY_RECODED, Year, left_right, ratio_leftright, logit_left_right, classic_economic, 
       ratio_economic, logit_economic, classic_gal_tan, ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, 
       ratio_economic_gal_tan, logit_economic_gal_tan)

Legislature_RECODED PARTY_RECODED Year left_right ratio_leftright
1 8                   DP            1979 -41.420    -0.8716330     
2 8                   DP            1980 -29.035    -0.7395584     
3 8                   DP            1981 -20.260    -0.4877227     
4 8                   DP            1982 -22.310    -0.5803005     
  logit_left_right classic_economic ratio_economic logit_economic
1 -2.539048        -29.79           -1.0000000     -4.103965     
2 -2.095848        -23.14           -0.8963370     -2.647445     
3 -1.036266        -15.24           -0.7912773     -1.956103     
4 -1.349249        -18.91           -0.6588462     -1.860461     
  classic_gal_tan ratio_gal_tan logit_gal_tan classic_economic_gal_tan
1  0.000          NaN            0.0000000     0.000                  
2 -0.315           -1           -0.4076824    -0.315                  
3 -5.420           -1           -2.4714836    -5.420                  
4 -0.770           -1           -0.8771326    -1.485                  
  ratio_economic_gal_tan logit_economic_gal_tan
1        NaN              0.0000000            
2 -1.0000000             -0.4076824            
3 -1.0000000             -2.4714836            
4 -0.8565737             -1.1686617

In [74]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ -41.420,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -29.035,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -20.260,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -22.310,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ -0.8716330,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -0.7395584,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -0.4877227,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -0.5803005,
TRUE ~ ratio_leftright),
      logit_left_right = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ -2.539048,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -2.095848,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -1.036266,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -1.349249,
TRUE ~ logit_left_right),
      classic_economic = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ -29.79,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -23.14,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -15.24,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -18.91,
TRUE ~ classic_economic),
      ratio_economic = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ -1.0000000,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -0.8963370,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -0.7912773,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -0.6588462,
TRUE ~ ratio_economic),
      logit_economic = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ -4.103965,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -2.647445,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -1.956103,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -1.860461,
TRUE ~ logit_economic),
      classic_gal_tan = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ 0.000,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -0.315,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -5.420,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -0.770,
TRUE ~ classic_gal_tan),
      ratio_gal_tan = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -1,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -1,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -1,
TRUE ~ ratio_gal_tan),
      logit_gal_tan = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ 0.0000000,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -0.4076824,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -2.4714836,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -0.8771326,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ 0.000,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -0.315,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -5.420,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -1.485,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ -1.0000000,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -1.0000000,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -0.8565737,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1979 ~ 0.0000000,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1980 ~ -0.4076824,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1981 ~ -2.4714836,
gruppoP_recoded == "PDUP" & legislature == 8 & year == 1982 ~ -1.1686617,
TRUE ~ logit_economic_gal_tan)) 

## Sinistra Indipendente (SI) (ninth legislature)

The Independent Left was created by the Italian Communist Party with the goal to reinforce its leadership over the Italian left after the passage of the Italian Socialist Party to an alliance with the centrist Italian Christian Democracy. The values pertaining to PCI will be used to fill the NA values for SI.

In [82]:
ilsd %>% 
filter(Legislature_RECODED == 9 & PARTY_RECODED == "PCI") %>% 
select(Legislature_RECODED, PARTY_RECODED, Year, left_right, ratio_leftright, logit_left_right, classic_economic, 
       ratio_economic, logit_economic, classic_gal_tan, ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, 
       ratio_economic_gal_tan, logit_economic_gal_tan)

Legislature_RECODED PARTY_RECODED Year left_right ratio_leftright
1 9                   PCI           1983 -26.17944  -0.6308939     
2 9                   PCI           1986 -17.53310  -0.5499507     
3 9                   PCI           1987 -17.83554  -0.6542756     
  logit_left_right classic_economic ratio_economic logit_economic
1 -1.437221        -16.44706        -0.7288902     -1.726618     
2 -1.189282        -11.80513        -0.6850729     -1.541955     
3 -1.486563        -14.19994        -1.0000000     -3.380991     
  classic_gal_tan ratio_gal_tan logit_gal_tan classic_economic_gal_tan
1 -2.911066       -1            -1.9201721    -2.911066               
2 -0.197665       -1            -0.3331309    -0.197665               
3 -2.736114       -1            -1.8675205    -2.736114               
  ratio_economic_gal_tan logit_economic_gal_tan
1 -1                     -1.9201721            
2 -1                     -0.3331309            
3 -1                     -1.8675205

In [84]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -26.17944,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -17.53310,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -17.83554,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -0.6308939,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -0.5499507,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -0.6542756,
TRUE ~ ratio_leftright),
      logit_left_right = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -1.437221,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -1.189282,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -1.486563,
TRUE ~ logit_left_right),
      classic_economic = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -16.44706,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -11.80513,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -14.19994,
TRUE ~ classic_economic),
      ratio_economic = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -0.7288902,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -0.6850729,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -1.0000000,
TRUE ~ ratio_economic),
      logit_economic = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -1.726618,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -1.541955,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -3.380991,
TRUE ~ logit_economic),
      classic_gal_tan = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -2.911066,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -0.197665,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -2.736114,
TRUE ~ classic_gal_tan),
      ratio_gal_tan = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -1,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -1,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -1,
TRUE ~ ratio_gal_tan),
      logit_gal_tan = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -1.9201721,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -0.3331309,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -1.8675205,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -2.911066,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -0.197665,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -2.736114,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -1,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -1,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -1,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded == "SI" & legislature == 9 & year == 1983 ~ -1.9201721,
gruppoP_recoded == "SI" & legislature == 9 & year == 1986 ~ -0.3331309,
gruppoP_recoded == "SI" & legislature == 9 & year == 1987 ~ -1.8675205,
TRUE ~ logit_economic_gal_tan)) 

## Sinistra Indipendente (SIN/IND/) (tenth legislature) 

The Independent Left was created by the Italian Communist Party with the goal to reinforce its leadership over the Italian left after the passage of the Italian Socialist Party to an alliance with the centrist Italian Christian Democracy. The values pertaining to PCI will be used to fill the NA values for SI.

In [95]:
ilsd %>% 
filter(Legislature_RECODED == 10 & PARTY_RECODED == "PCI") %>% 
select(Legislature_RECODED, PARTY_RECODED, Year, left_right, ratio_leftright, logit_left_right, classic_economic, 
       ratio_economic, logit_economic, classic_gal_tan, ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, 
       ratio_economic_gal_tan, logit_economic_gal_tan)

Legislature_RECODED PARTY_RECODED Year left_right ratio_leftright
1 10                  PCI           1987 -32.973785 -0.7027305     
2 10                  PCI           1988  -7.032656 -0.2811525     
3 10                  PCI           1989  -3.702825 -0.1540813     
  logit_left_right classic_economic ratio_economic logit_economic
1 -1.6885495       -18.069944       -0.8357102     -2.190400     
2 -0.5544711        -3.094237       -0.9084395     -1.743499     
3 -0.2980383        -7.299153       -0.8076959     -1.845620     
  classic_gal_tan ratio_gal_tan logit_gal_tan classic_economic_gal_tan
1 -0.1333333      -1            -0.2363888    -0.1333333              
2 -0.1932203      -1            -0.3267398    -0.1932203              
3 -2.0819774      -1            -1.6417027    -2.0469491              
  ratio_economic_gal_tan logit_economic_gal_tan
1 -1.0000000             -0.2363888            
2 -1.0000000             -0.3267398            
3 -0.9669077             -1.5739913

In [102]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -32.973785,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -7.032656,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -3.702825,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -0.7027305,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -0.2811525,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -0.1540813,
TRUE ~ ratio_leftright),
      logit_left_right = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -1.6885495,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -0.5544711,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -0.2980383,
TRUE ~ logit_left_right),
      classic_economic = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -18.069944,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -3.094237,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -7.299153,
TRUE ~ classic_economic),
      ratio_economic = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -0.8357102,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -0.9084395,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -0.8076959,
TRUE ~ ratio_economic),
      logit_economic = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -2.190400,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -1.743499,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -1.845620,
TRUE ~ logit_economic),
      classic_gal_tan = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -0.1333333,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -0.1932203,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -2.0819774,
TRUE ~ classic_gal_tan),
      ratio_gal_tan = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -1,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -1,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -1,
TRUE ~ ratio_gal_tan),
      logit_gal_tan = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -0.2363888,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -0.3267398,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -1.6417027,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -0.1333333,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -0.1932203,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -2.0469491,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -1.0000000,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -1.0000000,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -0.9669077,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1987 ~ -0.2363888,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1988 ~ -0.3267398,
gruppoP_recoded == "SIN/IND/" & legislature == 10 & year == 1989 ~ -1.5739913,
TRUE ~ logit_economic_gal_tan)) 

## PDS

Founded in February 1991 as the post-communist evolution of PCI

In [95]:
ilsd %>% 
filter(Legislature_RECODED == 10 & PARTY_RECODED == "PCI") %>% 
select(Legislature_RECODED, PARTY_RECODED, Year, left_right, ratio_leftright, logit_left_right, classic_economic, 
       ratio_economic, logit_economic, classic_gal_tan, ratio_gal_tan, logit_gal_tan, classic_economic_gal_tan, 
       ratio_economic_gal_tan, logit_economic_gal_tan)

Legislature_RECODED PARTY_RECODED Year left_right ratio_leftright
1 10                  PCI           1987 -32.973785 -0.7027305     
2 10                  PCI           1988  -7.032656 -0.2811525     
3 10                  PCI           1989  -3.702825 -0.1540813     
  logit_left_right classic_economic ratio_economic logit_economic
1 -1.6885495       -18.069944       -0.8357102     -2.190400     
2 -0.5544711        -3.094237       -0.9084395     -1.743499     
3 -0.2980383        -7.299153       -0.8076959     -1.845620     
  classic_gal_tan ratio_gal_tan logit_gal_tan classic_economic_gal_tan
1 -0.1333333      -1            -0.2363888    -0.1333333              
2 -0.1932203      -1            -0.3267398    -0.1932203              
3 -2.0819774      -1            -1.6417027    -2.0469491              
  ratio_economic_gal_tan logit_economic_gal_tan
1 -1.0000000             -0.2363888            
2 -1.0000000             -0.3267398            
3 -0.9669077             -1.5739913

In [115]:
joined_texts <- joined_texts %>% 
mutate(left_right = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -32.973785,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -7.032656,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -3.702825,
TRUE ~ left_right),
      ratio_leftright = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -0.7027305,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -0.2811525,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -0.1540813,
TRUE ~ ratio_leftright),
      logit_left_right = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -1.6885495,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -0.5544711,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -0.2980383,
TRUE ~ logit_left_right),
      classic_economic = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -18.069944,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -3.094237,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -7.299153,
TRUE ~ classic_economic),
      ratio_economic = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -0.8357102,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -0.9084395,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -0.8076959,
TRUE ~ ratio_economic),
      logit_economic = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -2.190400,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -1.743499,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -1.845620,
TRUE ~ logit_economic),
      classic_gal_tan = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -0.1333333,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -0.1932203,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -2.0819774,
TRUE ~ classic_gal_tan),
      ratio_gal_tan = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -1,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -1,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -1,
TRUE ~ ratio_gal_tan),
      logit_gal_tan = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -0.2363888,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -0.3267398,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -1.6417027,
TRUE ~ logit_gal_tan),
       classic_economic_gal_tan = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -0.1333333,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -0.1932203,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -2.0469491,
TRUE ~ classic_economic_gal_tan),
       ratio_economic_gal_tan = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -1.0000000,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -1.0000000,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -0.9669077,
TRUE ~ ratio_economic_gal_tan),
       logit_economic_gal_tan = case_when(
gruppoP_recoded == "PDS" & legislature == 10 & year == 1987 ~ -0.2363888,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1988 ~ -0.3267398,
gruppoP_recoded == "PDS" & legislature == 10 & year == 1989 ~ -1.5739913,
TRUE ~ logit_economic_gal_tan)) 

In [117]:
joined_texts %>% filter(is.na(left_right)) %>% distinct(gruppoP_recoded)

legislature
1   1         
2   2         
3   3         
4   4         
5   5         
6   6         
7   7         
8   8         
9   9         
10 10         
11 11         
12 11         
13 12         
14 12         
15 12         
16 12         
17 13         
18 13         
19 13         
20 13         
21 13         
22 14         
23 15         
24 15         
25 15         
26 16         
27 16         
28 16         
29 17         
30 17         
31 17         
32 18         
33 18         
   gruppoP_recoded                                                                                                                                             
1  MISTO                                                                                                                                                       
2  MISTO                                                                                                                                                       
3  MISTO                                                                                                                                                       
4  MISTO                                                                                                                                                       
5  MISTO                                                                                                                                                       
6  MISTO                                                                                                                                                       
7  MISTO                                                                                                                                                       
8  MISTO                                                                                                                                                       
9  MISTO                                                                                                                                                       
10 MISTO                                                                                                                                                       
11 MISTO                                                                                                                                                       
12 UDC                                                                                                                                                         
13 DEMO                                                                                                                                                        
14 MISTO                                                                                                                                                       
15 FLD                                                                                                                                                         
16 LIFED                                                                                                                                                       
17 PD-ULIVO                                                                                                                                                    
18 UDEUR                                                                                                                                                       
19 MISTO                                                                                                                                                       
20 UDR                                                                                                                                                         
21 DEM-U                                                                                                                             

## UDC

In [125]:
ilsd %>% 
filter(Legislature_RECODED == 11) %>% 
arrange(PARTY) %>% 
distinct(PARTY)

PARTY   Legislature_RECODED PARTY_RECODED Year
1  CABINET 11                  CABINET       1992
2  CABINET 11                  CABINET       1993
3  DC      11                  DC            1992
4  DC      11                  DC            1993
5  LN      11                  LEGA          1992
6  LN      11                  LEGA          1993
7  MSI     11                  MSI           1992
8  MSI     11                  MSI           1993
9  PDS     11                  PDS           1992
10 PDS     11                  PDS           1993
11 PLI     11                  PLI           1992
12 PLI     11                  PLI           1993
13 PR      11                  RADICALE      1992
14 PR      11                  RADICALE      1993
15 PRI     11                  PRI           1992
16 PRI     11                  PRI           1993
17 PSDI    11                  PSDI          1992
18 PSDI    11                  PSDI          1993
19 PSI     11                  PSI           1992
20 PSI     11                  PSI           1993
21 RC      11                  RC            1992
22 RC      11                  RC            1993
23 RETE    11                  RETE          1992
24 RETE    11                  RETE          1993
25 VER     11                  VERDI         1992
26 VER     11                  VERDI         1993

In [123]:
joined_texts %>% 
filter(legislature == 11) %>% 
arrange(gruppoP) %>% 
distinct(gruppoP)

legislature gruppoP  gruppoP_recoded
1  11          CCD      UDC            
2  11          DC-PPI   DC             
3  11          FED/EUR/ RADICALE       
4  11          LEGA-N   LEGA           
5  11          MISTO    MISTO          
6  11          MSI-DN   MSI            
7  11          PDS      PDS            
8  11          PLI      PLI            
9  11          PRI      PRI            
10 11          PSDI     PSDI           
11 11          PSI      PSI            
12 11          RC       RC             
13 11          RETE     RETE           
14 11          VERDI    VERDI

In [120]:
joined_texts %>% 
filter(legislature == 11 & gruppoP_recoded == "UDC") %>% 
distinct(year)

year legislature gruppoP_recoded
1 1992 11          UDC            
2 1993 11          UDC            
3 1994 11          UDC

## Filling down the remaining null values

Filling down the remaining values

In [116]:
joined_texts <- joined_texts %>% 
arrange(year) %>% 
group_by(gruppoP_recoded, legislature) %>% 
fill_(fill_cols = c("left_right", "ratio_leftright", "logit_left_right", "classic_economic", "ratio_economic", 
                    "logit_economic", "classic_gal_tan", "ratio_gal_tan", "logit_gal_tan", "classic_economic_gal_tan", 
                    "ratio_economic_gal_tan", "logit_economic_gal_tan"),
       .direction = "down")